vaultutil/pkg/vault/manager.go

```diff
@@ -0,0 +1,365 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/puppetlabs/leg/vaultutil/pkg/model"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type VaultSystemManager struct {
+	kubeClient  client.Client
+	vaultClient *vaultapi.Client
+	vaultConfig *VaultConfig
+}
+
+func (vsm *VaultSystemManager) Initialize(ctx context.Context) (*model.VaultKeys, error) {
```

Comment:
It may be worth documenting somewhere how specific this is to Kubernetes, e.g. that this exists to set up a Vault instance that is authenticated with a Kubernetes secret, since from the name alone (without reading all the code) it wasn't totally clear that it was requirement. Or maybe change the name to something like `InitializeWithKubernetesSecretAccess` or something?

Reply:
I'd like to split this up into two different managers in the future. One that only wraps the Vault calls with a clean interface, and a higher level manager that has the opinionated use of Kubernetes secrets, service accounts, etc.

Currently, I'd like to avoid bleeding implementation details by changing the name to specify it. That would then suggest that the secret object metadata should be either passed in as parameters (or likely added as options).

Several of these calls have internal mechanics that are not clear from the interface alone (which has both positives and negatives). These will either be transitioned into a very-specific higher level implementation with a name that suggests it's specific Kubernetes use, or update the interface to specific the explicit details as parameters/options; potentially some of the logic my simply be removed in favor of the caller handling it directly instead.